orc8r/cloud/go/obsidian/swagger/swagger_spec_client.go

```diff
@@ -0,0 +1,59 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger
+
+import (
+	"context"
+	"strings"
+
+	"magma/orc8r/cloud/go/obsidian/swagger/protos"
+	merrors "magma/orc8r/lib/go/errors"
+	"magma/orc8r/lib/go/registry"
+
+	"github.com/golang/glog"
+)
+
+type swaggerClient struct {
+	// service name of the swaggerClient
+	// should always be lowercase to match service registry convention
+	service string
+}
+
+func NewSwaggerClient(serviceName string) SwaggerClient {
+	return &swaggerClient{service: strings.ToLower(serviceName)}
+}
+
+func (s *swaggerClient) GetSpec() (string, error) {
+	c, err := s.GetSwaggerSpecClient()
+	if err != nil {
+		return "", err
+	}
+
+	res, err := c.GetSpec(context.Background(), &protos.GetSpecRequest{})
+	if err != nil {
+		return "", err
+	}
+
+	return res.SwaggerSpec, nil
+}
+
+func (s *swaggerClient) GetSwaggerSpecClient() (protos.SwaggerSpecClient, error) {
```

Comment:
- You're exporting a method that can only ever be called by the object itself -- but the object is private and the New method returns the iface. So something's wrong
- In this case, the answer to the issue is this method should be private